src/App.tsx

```diff
@@ -8,35 +8,100 @@ import { PostsList } from './components/PostsList';
+  const [selectedPostId, setSelectedPostId] = useState<SelectedPostId>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const handleUserSelect = async (userId: number) => {
+    setSelectedUserId(userId);
+    setIsLoading(true);
+    setError(false);
+
+    try {
+      const loadedPosts = await getUserPosts(userId);
+
+      setPosts(loadedPosts);
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handlePostSelect = (postId: SelectedPostId) =>
+    setSelectedPostId(postId);
+
+  const loadUsers = async () => {
+    try {
+      const loadedUsers = await getUsers();
+
+      setUsers(loadedUsers);
+    } catch {
+      setError(true);
+    }
+  };
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  const selectedUser = useMemo(
+    () => users.find(user => user.id === selectedUserId) || null,
+    [users, selectedUserId],
+  );
+
+  const selectedPost = useMemo(
+    () => posts.find(post => post.id === selectedPostId) || null,
+    [posts, selectedPostId],
+  );
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  onUserSelect={handleUserSelect}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
-
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
-
-                <PostsList />
+                {isLoading ? (
+                  <Loader />
+                ) : error ? (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                ) : !selectedUser ? (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                ) : !posts.length ? (
+                  <div className="notification is-warning" data-cy="NoPostsYet">
+                    No posts yet
+                  </div>
+                ) : (
+                  <PostsList
+                    posts={posts}
+                    selectedPostId={selectedPostId}
+                    onPostSelect={handlePostSelect}
+                  />
+                )}
```

Comment:
Extensive use of nested ternary operators can make the code hard to read and maintain. It's often better to use if-else statements.